app/models/deploy_service.rb

```diff
@@ -19,7 +19,13 @@ def deploy!(stage, attributes)
   def confirm_deploy!(deploy)
     send_before_notifications(deploy)
 
-    job_execution = JobExecution.start_job(deploy.reference, deploy.job)
+    stage = deploy.stage
+
+    job_execution = JobExecution.start_job(
+      deploy.reference, deploy.job,
+      construct_env(stage).merge(key: stage.id)
+    )
+
     send_sse_deploy_update('start', deploy)
 
     job_execution.on_complete do
```

Comment:
Good call, I wonder if this and the subscriber errors may be why we've seen issues where emails and notifications aren't sent out